app/controllers/schools/downloads_controller.rb

```diff
@@ -1,8 +1,18 @@
 module Schools
   class DownloadsController < ApplicationController
     load_and_authorize_resource :school
+    before_action :authorized?
 
     def index
     end
+
+    private
+
+    def authorized?
+      unless can?(:download_school_data, @school)
+        flash[:error] = "You are not authorized to view that page."
+        redirect_to root_path
```

Comment:
Maybe redirect to school dashboard instead of home page?

Reply:
👍